packages/proxy/lib/http/index.ts

```diff
@@ -182,7 +183,14 @@ export function _runStage (type: HttpStages, ctx: any, onError: Function) {
       const fullCtx = {
         next: () => {
           fullCtx.next = () => {
-            throw new Error('Error running proxy middleware: Cannot call this.next() more than once in the same middleware function. Doing so can cause unintended issues.')
+            const error = new Error('Error running proxy middleware: Detected `this.next()` was called more than once in the same middleware function, but a middleware can only be completed once.')
+
+            if (ctx.error) {
+              error.message = error.message += '\nThis middleware invocation previously encountered an error which may be related, see `error.cause`'
+              error['cause'] = ctx.error
```

Comment:
I think the type error this works around could be due to us using an outdated `@types/node`... but that may be more than a 1-line change to update.

It doesn't look like `logError` will log `cause`, so I'm not sure this is going to get logged by line 218: https://github.com/cypress-io/cypress/blob/07052d82a016c16af42d23363e359c9a68bb6123/packages/errors/src/errorUtils.ts#L55-L72

Reply:
Took a look at updating the `@types/node` bindings - it would end up touching a 3-4 other packages and my `yarn` was having serious indigestion trying to bump the version so I may leave as-is with the type workaround just to keep this PR targeted